packages/protocol/contracts/psp/AirnodePsp.sol

```diff
@@ -0,0 +1,109 @@
+    struct Subscription {
+        bytes32 templateId;
+        address conditionAddress;
+        bytes4 conditionFunctionId;
+        address fulfillAddress;
+        bytes4 fulfillFunctionId;
+        bytes parameters;
+    }
+
+    IAirnodeRrp public airnodeRrp;
+
+    mapping(bytes32 => Subscription) public subscriptions;
+
+    constructor(address airnodeRrp_) {
+        airnodeRrp = IAirnodeRrp(airnodeRrp_);
+    }
+
+    // We store subscriptions in storage because we don't trust event logs to persist.
+    // A subscription should be able to live for years when left untouched.
+    function createSubscription(
+        bytes32 templateId,
+        address conditionAddress,
+        bytes4 conditionFunctionId,
+        address fulfillAddress,
+        bytes4 fulfillFunctionId,
+        bytes calldata parameters
+    ) external returns (bytes32 subscriptionId) {
+        // Subscription IDs do not need to be unique. Airnode will not serve the same
+        // sbuscription ID multiple times
+        subscriptionId = keccak256(
+            abi.encode(
+                templateId,
+                conditionAddress,
+                conditionFunctionId,
+                fulfillAddress,
+                fulfillFunctionId,
+                parameters
+            )
+        );
+        subscriptions[subscriptionId] = Subscription({
+            templateId: templateId,
+            conditionAddress: conditionAddress,
+            conditionFunctionId: conditionFunctionId,
+            fulfillAddress: fulfillAddress,
+            fulfillFunctionId: fulfillFunctionId,
+            parameters: parameters
+        });
+    }
+
+    // We are having the node fulfill through this function because we don't want the requester
+    // contract to have to import ECDSA.sol
+    // The node doesn't attempt to fulfill if `callSuccess` is false, logs the revert string internally.
+    function fulfill(
+        bytes32 subscriptionId,
+        uint256 timestamp,
+        bytes calldata data,
+        bytes calldata signature
+    ) external returns (bool callSuccess, bytes memory callData) {
+        Subscription storage subscription = subscriptions[subscriptionId];
+        // Note that we are not verifying that msg.sender is a pre-determined `sponsorWallet`.
+        // This because one cannot "botch" (e.g., by specifying a low gas limit) a PSP fulfillment
+        // like an RRP fulfillment (because previous failed fulfillments don't prevent future
+        // successful ones).
+        (address airnode, , ) = airnodeRrp.templates(subscription.templateId);
+        // It is a concern for someone to have the API provider sign a response, to be used much
+        // later (for example, would be very problematic for asset price data feeds). Therefore,
+        // we add the timestamp to the hash to be signed. 
+        require(
+            (
+                keccak256(
+                    abi.encodePacked(subscriptionId, timestamp, data)
```

Comment:
I don't think I follow. Do you mean that a requester can save a timestamp when the request was made and then later once the Airnode fulfill transaction is received it will reject, because the timestamp on the Airnode response transaction is much larger? (a lot of time has passed)

Because I don't see a problem with this requester behaviour.